### DISCUSSION TITLE
Auto "enforce string" for properties where schema expects string

### DISCUSSION THREAD

**Discussion by RyKilleen** [josdejong/svelte-jsoneditor#179] [Ideas]
Use case:

We have a config where several values have to be numbers _saved as strings_. The JSON Editor recognizes it as a number and saves it as a number (appropriately so!) Our users need to edit each value and select "Enforce String". (Thankful for this escape hatch which critically unblocks us.)

It's currently tedious for our users to check "Enforce string" for each one of these values, and our schema strongly enforces their value as a string. Not sure if it is even feasible architecturally for the JSON editor, but would be incredibly valuable for us.

If there are any other workarounds / options available in the editor I'm happy to pursue those as well!

**Comment by josdejong** [comment]
Thanks for your suggestion Ryan. I like the idea of utilizing knowledge from a JSON schema to specify the right type of a value (number or string containing a number)!

Note that you can solve this issue for your users without needing an update in `svelte-jsoneditor` itself. You can write a function which validates the config document, checks the type of the value of these specific fields, and if needed change them into a string. I think this function could run on every change of the document contents, immediately correcting the contents. This could be implemented specifically for your config, or more generic, based on a JSON schema. What do you think?

On a side note: I'm not entirely happy with the current implementation of the "Enforce string" feature and have some ideas to improve on this, see https://github.com/josdejong/svelte-jsoneditor/issues/104#issuecomment-1169663562.

**Reply by RyKilleen** [reply]
I think for our purpose, in a very large JSON object with a pre-defined schema and many, many fields it might not be tenable to manually convert specific fields, unless I'm misunderstanding!

**Reply by josdejong** [reply]
I'm not sure what you mean with "manually". I think least the users should not be bothered with this. I meant doing some post-processing of the document: write a little script that iterates over the document, verifies the type of the fields, and correct them where needed.

BTW help implementing this feature would be welcome. I expect this to be not _that_ complicated, and there is some comparable logic with the enum dropdown that can help figure out how to approach this.
